src/Formik.tsx

```diff
@@ -665,6 +665,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
             .catch(_errors => {
               if (!!isMounted.current) {
                 dispatch({ type: 'SUBMIT_FAILURE' });
+                throw _errors instanceof Error ? _errors : new Error(_errors);
```

Comment:
Can't we just rethrow directly? If user throw weird things, he should receive weird things in catch imo